engine/flink/management/periodic/src/main/scala/pl/touk/nussknacker/engine/management/periodic/PeriodicStateStatus.scala

```diff
@@ -19,7 +20,7 @@ object PeriodicStateStatus {
   }
 
   val statusActionsPF: PartialFunction[StateStatus, List[ProcessActionType]] = {
-    case _: RunningStateStatus => List(ProcessActionType.Cancel) //periodic processes cannot be redeployed from GUI
+    case s: StateStatus if s.name == SimpleStateStatus.Running.name => List(ProcessActionType.Cancel) //periodic processes cannot be redeployed from GUI
```

Comment:
What if somebody creates a status finished with the same name?

Reply:
Now we can have multiple pluggable deployment managers, each of them is allowed to define its statuses, event with the same names. Each manager is responsible for its statuses and knows how to handle them, so the same name is not a problem here. It becomes a problem when we need to provide status information to the user (e.g via scenarios panel). Here we have an assertion that states have unique definitions, and are distinct. It is checked in `ProcessStateDefinitionService`.

Reply:
@lciolecki We are planning to remove all `isXYZ` from `StateStatus`. It introduced a lot of confusion: when sth is running, when is not? When is failed when not? We find out that we can reduce all places of `isXYZ` usages into DeploymentManager's logic, which is customizable / pluggable, so developer of extensions will know best, how to handle each state. In this case `PeriodicDeploymentManager` is tightly coupled with `FlinkDeploymentManager`, so he know best that it will only one representation of `RunningStateStatus`, so he don't need to check `isRunning`. BTW by checking `isRunning == true` @gskrobisz would introduce a regression back, because `ScheduledStatus` isRunning = true, which is a great example of that those `isXYZ` methods are confusing :P